### ISSUE TITLE
[java] SwitchStmtsShouldHaveDefault triggered when default case is present

### ISSUE THREAD

**Issue by karel1980** [pmd/pmd#3605]
**Affects PMD Version:** 
6.40.0

**Rule:**
PMD.SwitchStmtsShouldHaveDefault
https://pmd.github.io/pmd-6.40.0/pmd_rules_java_bestpractices.html#switchstmtsshouldhavedefault

**Description:**

**Code Sample demonstrating the issue:**

```java
enum State {
  ON, OFF
}

... switch(state) {
      case ON -> System.out.println("it is on");
      case OFF -> System.out.println("it is off");
      default -> System.out.println("it is neither on nor off - should not happen? maybe null?")
    }
```

**Expected outcome:**
No violation

** Actual outcome:**
PMD reports a violation at line `switch(state)`, but that's wrong. That's a false positive.

**Running PMD through:** *Gradle*
